SU2_CFD/include/numerics/scalar/scalar_convection.hpp

```diff
@@ -130,9 +131,14 @@ class CUpwScalar : public CNumerics {
         q_ij += 0.5 * (V_i[iDim + idx.Velocity()] + V_j[iDim + idx.Velocity()]) * Normal[iDim];
       }
     }
-
-    a0 = 0.5 * (q_ij + fabs(q_ij));
-    a1 = 0.5 * (q_ij - fabs(q_ij));
+    
+    if(config->GetKind_Upwind_Species() == UPWIND::BOUNDED_SCALAR){
```

Comment:
This needs to be a boolean set by the derived class (either turbulence or passive scalar, otherwise you cannot change them independently).